install_sct.bat

```diff
@@ -61,7 +61,7 @@ if exist requirements-freeze.txt (
 )
 echo:
 echo ### Installing SCT and its dependencies from %requirements_file%...
-rem Pip needs to be upgraded because default p3.7 pip won't resolve dependency conflicts correctly
+rem Pip needs to be upgraded because default p3.8 pip won't resolve dependency conflicts correctly

```

Comment:
Is this still true with Python 3.8? I don't have a Windows machine handy to test it.

Reply:
My apologies! The comment I wrote here does a poor job at actually documenting the issue. 

> _(What exactly **is** the dependency resolution issue? Which versions of `pip` have this dependency resolution issue?_
> _My comment should taken the time to answer these questions...)_

There are no clues in [the original `install_sct.bat` issue](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3682#issuecomment-1034039052), because the comment is written in the exact same way. 

So, all we have to go off of is this minimally-reproducible example from the link:

```bash
cd <sct-repo-clone-location>
python -m venv venv_sct
cd venv_sct/Scripts
activate
# upgrade because default p3.7 pip won't resolve dependency conflicts correctly
python -m pip install --upgrade pip setuptools wheel
cd ../..
pip install -r requirements.txt
```

I ran this example for 4 different cases:

<details>
<summary> Python 3.7.9, pip 20.1.1 (no upgrade) </summary>

```console
venv_sct@C:\Users\joshu:repos\spinalcordtoolbox$ pip install -r requirements.txt
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cpu
Collecting dipy
  Using cached dipy-1.5.0-cp37-none-win_amd64.whl (7.3 MB)
Collecting ivadomed~=2.9.6
  Downloading ivadomed-2.9.7-py3-none-any.whl (231 kB)
     |████████████████████████████████| 231 kB 3.3 MB/s
Collecting matplotlib
  Downloading matplotlib-3.5.3-cp37-cp37m-win_amd64.whl (7.2 MB)
     |████████████████████████████████| 7.2 MB 6.4 MB/s
Collecting msvc-runtime==14.29.30133
  Using cached msvc_runtime-14.29.30133-cp37-cp37m-win_amd64.whl (1.8 MB)
Collecting nibabel
  Downloading nibabel-4.0.2-py3-none-any.whl (3.3 MB)
     |████████████████████████████████| 3.3 MB 6.4 MB/s
Collecting nilearn
  Downloading nilearn-0.9.2-py3-none-any.whl (9.6 MB)
     |████████████████████████████████| 9.6 MB 6.8 MB/s
Collecting numpy
  Downloading numpy-1.21.6-cp37-cp37m-win_amd64.whl (14.0 MB)
     |████████████████████████████████| 14.0 MB 6.8 MB/s
Collecting onnxruntime>=1.7.0
  Downloading onnxruntime-1.13.1-cp37-cp37m-win_amd64.whl (5.9 MB)
     |████████████████████████████████| 5.9 MB 6.4 MB/s
Collecting pandas
  Using cached pandas-1.3.5-cp37-cp37m-win_amd64.whl (10.0 MB)
Collecting portalocker
  Downloading portalocker-2.6.0-py2.py3-none-any.whl (15 kB)
Collecting psutil
  Downloading psutil-5.9.4-cp36-abi3-win_amd64.whl (252 kB)
     |████████████████████████████████| 252 kB 6.8 MB/s
Collecting pyqt5<5.15.0,>=5.12.0
  Downloading PyQt5-5.14.2-5.14.2-cp35.cp36.cp37.cp38-none-win_amd64.whl (52.9 MB)
     |████████████████████████████████| 52.9 MB 159 kB/s
Collecting pytest
  Downloading pytest-7.2.0-py3-none-any.whl (316 kB)
     |████████████████████████████████| 316 kB ...
Collecting pytest-cov
  Downloading pytest_cov-4.0.0-py3-none-any.whl (21 kB)
Collecting raven
  Using cached raven-6.10.0-py2.py3-none-any.whl (284 kB)
Collecting requests
  Downloading requests-2.28.1-py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 654 kB/s
Collecting requirements-parser
  Using cached requirements_parser-0.5.0-py3-none-any.whl (18 kB)
Collecting scipy
  Using cached scipy-1.7.3-cp37-cp37m-win_amd64.whl (34.1 MB)
Collecting scikit-image
  Downloading scikit_image-0.19.3-cp37-cp37m-win_amd64.whl (12.1 MB)
     |████████████████████████████████| 12.1 MB 3.3 MB/s
Collecting scikit-learn
  Using cached scikit_learn-1.0.2-cp37-cp37m-win_amd64.whl (7.1 MB)
Collecting xlwt
  Using cached xlwt-1.3.0-py2.py3-none-any.whl (99 kB)
Collecting tqdm
  Downloading tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
     |████████████████████████████████| 78 kB 3.4 MB/s
Collecting transforms3d
  Downloading transforms3d-0.4.1.tar.gz (1.4 MB)
     |████████████████████████████████| 1.4 MB 6.4 MB/s
Collecting urllib3[secure]
  Downloading urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
     |████████████████████████████████| 140 kB 6.4 MB/s
Collecting pytest_console_scripts
  Using cached pytest_console_scripts-1.3.1-py3-none-any.whl (9.0 kB)
Collecting pyyaml
  Using cached PyYAML-6.0-cp37-cp37m-win_amd64.whl (153 kB)
Collecting voxelmorph
  Downloading voxelmorph-0.2-py3-none-any.whl (54 kB)
     |████████████████████████████████| 54 kB 1.2 MB/s
Collecting wquantiles
  Using cached wquantiles-0.6-py3-none-any.whl (3.3 kB)
Collecting h5py>=2.8.0
  Downloading h5py-3.7.0-cp37-cp37m-win_amd64.whl (2.6 MB)
     |████████████████████████████████| 2.6 MB 6.4 MB/s
Collecting torchvision<=0.12.0,>=0.9.1
  Downloading torchvision-0.12.0-cp37-cp37m-win_amd64.whl (1.0 MB)
     |████████████████████████████████| 1.0 MB 6.4 MB/s
Collecting csv-diff>=1.0
  Using cached csv_diff-1.1-py3-none-any.whl (12 kB)
Collecting torchio>=0.18.68
  Downloading torchio-0.18.86-py2.py3-none-any.whl (172 kB)
     |████████████████████████████████| 172 kB 3.3 MB/s
Collecting imageio~=2.19
  Downloading imageio-2.22.4-py3-none-any.whl (3.4 MB)
     |████████████████████████████████| 3.4 MB 6.4 MB/s
Collecting pybids>=0.14.0
  Downloading pybids-0.15.5-py3-none-any.whl (14.9 MB)
     |████████████████████████████████| 14.9 MB 3.2 MB/s
Collecting torch<=1.11.0,>=1.8.1
  Downloading torch-1.11.0-cp37-cp37m-win_amd64.whl (158.0 MB)
     |████████████████████████████████| 158.0 MB 31 kB/s
Collecting loguru~=0.5
  Using cached loguru-0.6.0-py3-none-any.whl (58 kB)
Collecting tensorboard>=1.15.0
  Downloading tensorboard-2.11.0-py3-none-any.whl (6.0 MB)
     |████████████████████████████████| 6.0 MB 6.8 MB/s
Collecting pyparsing<3,>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting joblib~=1.0
  Downloading joblib-1.2.0-py3-none-any.whl (297 kB)
     |████████████████████████████████| 297 kB 6.4 MB/s
Collecting wandb>=0.12.11
  Downloading wandb-0.13.5-py2.py3-none-any.whl (1.9 MB)
     |████████████████████████████████| 1.9 MB ...
Collecting seaborn~=0.11
  Downloading seaborn-0.12.1-py3-none-any.whl (288 kB)
     |████████████████████████████████| 288 kB ...
Collecting fonttools>=4.22.0
  Downloading fonttools-4.38.0-py3-none-any.whl (965 kB)
     |████████████████████████████████| 965 kB 6.8 MB/s
Collecting pillow>=6.2.0
  Downloading Pillow-9.3.0-1-cp37-cp37m-win_amd64.whl (2.5 MB)
     |████████████████████████████████| 2.5 MB 6.8 MB/s
Collecting kiwisolver>=1.0.1
  Downloading kiwisolver-1.4.4-cp37-cp37m-win_amd64.whl (54 kB)
     |████████████████████████████████| 54 kB 4.1 MB/s
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Requirement already satisfied: setuptools in c:\users\joshu\repos\spinalcordtoolbox\venv_sct\lib\site-packages (from nibabel->-r requirements.txt (line 18)) (47.1.0)
Collecting lxml
  Downloading lxml-4.9.1-cp37-cp37m-win_amd64.whl (3.6 MB)
     |████████████████████████████████| 3.6 MB ...
Collecting protobuf
  Downloading protobuf-4.21.9-cp37-cp37m-win_amd64.whl (525 kB)
     |████████████████████████████████| 525 kB 6.8 MB/s
Collecting flatbuffers
  Downloading flatbuffers-22.11.23-py2.py3-none-any.whl (26 kB)
Collecting sympy
  Downloading sympy-1.10.1-py3-none-any.whl (6.4 MB)
     |████████████████████████████████| 6.4 MB ...
Collecting coloredlogs
  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
     |████████████████████████████████| 46 kB 85 kB/s
Collecting pytz>=2017.3
  Downloading pytz-2022.6-py2.py3-none-any.whl (498 kB)
     |████████████████████████████████| 498 kB 6.4 MB/s
Collecting pywin32>=226; platform_system == "Windows"
  Downloading pywin32-305-cp37-cp37m-win_amd64.whl (12.2 MB)
     |████████████████████████████████| 12.2 MB 3.3 MB/s
Collecting PyQt5-sip<13,>=12.7
  Downloading PyQt5_sip-12.11.0-cp37-cp37m-win_amd64.whl (77 kB)
     |████████████████████████████████| 77 kB 1.8 MB/s
Collecting exceptiongroup>=1.0.0rc8; python_version < "3.11"
  Downloading exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
Collecting tomli>=1.0.0; python_version < "3.11"
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting attrs>=19.2.0
  Downloading attrs-22.1.0-py2.py3-none-any.whl (58 kB)
     |████████████████████████████████| 58 kB 4.1 MB/s
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting colorama; sys_platform == "win32"
  Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting importlib-metadata>=0.12; python_version < "3.8"
  Downloading importlib_metadata-5.1.0-py3-none-any.whl (21 kB)
Collecting coverage[toml]>=5.2.1
  Downloading coverage-6.5.0-cp37-cp37m-win_amd64.whl (188 kB)
     |████████████████████████████████| 188 kB 6.4 MB/s
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 4.8 MB/s
Collecting certifi>=2017.4.17
  Downloading certifi-2022.9.24-py3-none-any.whl (161 kB)
     |████████████████████████████████| 161 kB 6.8 MB/s
Collecting charset-normalizer<3,>=2
  Downloading charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting types-setuptools>=57.0.0
  Downloading types_setuptools-65.6.0.1-py3-none-any.whl (48 kB)
     |████████████████████████████████| 48 kB 2.5 MB/s
Collecting tifffile>=2019.7.26
  Using cached tifffile-2021.11.2-py3-none-any.whl (178 kB)
Collecting networkx>=2.2
  Using cached networkx-2.6.3-py3-none-any.whl (1.9 MB)
Collecting PyWavelets>=1.1.1
  Using cached PyWavelets-1.3.0-cp37-cp37m-win_amd64.whl (4.2 MB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting pyOpenSSL>=0.14; extra == "secure"
  Downloading pyOpenSSL-22.1.0-py3-none-any.whl (57 kB)
     |████████████████████████████████| 57 kB 4.1 MB/s
Collecting cryptography>=1.3.4; extra == "secure"
  Downloading cryptography-38.0.4-cp36-abi3-win_amd64.whl (2.4 MB)
     |████████████████████████████████| 2.4 MB 6.8 MB/s
Collecting urllib3-secure-extra; extra == "secure"
  Downloading urllib3_secure_extra-0.1.0-py2.py3-none-any.whl (1.4 kB)
Collecting neurite>=0.2
  Downloading neurite-0.2-py3-none-any.whl (108 kB)
     |████████████████████████████████| 108 kB 6.4 MB/s
Collecting typing-extensions
  Downloading typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting dictdiffer
  Using cached dictdiffer-0.9.0-py2.py3-none-any.whl (16 kB)
Collecting click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting humanize
  Downloading humanize-4.4.0-py3-none-any.whl (106 kB)
     |████████████████████████████████| 106 kB 6.8 MB/s
Collecting SimpleITK!=2.0.*,!=2.1.1.1
  Downloading SimpleITK-2.2.0-cp37-cp37m-win_amd64.whl (27.1 MB)
     |████████████████████████████████| 27.1 MB 6.4 MB/s
Collecting Deprecated
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting typer[all]
  Downloading typer-0.7.0-py3-none-any.whl (38 kB)
Collecting bids-validator
  Downloading bids_validator-1.9.9-py2.py3-none-any.whl (21 kB)
Collecting num2words
  Downloading num2words-0.5.12-py3-none-any.whl (125 kB)
     |████████████████████████████████| 125 kB 6.4 MB/s
Collecting formulaic<0.4,>=0.2.4
  Downloading formulaic-0.3.4-py3-none-any.whl (68 kB)
     |████████████████████████████████| 68 kB 1.1 MB/s
Collecting sqlalchemy<1.4.0.dev0
  Using cached SQLAlchemy-1.3.24-cp37-cp37m-win_amd64.whl (1.2 MB)
Collecting win32-setctime>=1.0.0; sys_platform == "win32"
  Using cached win32_setctime-1.1.0-py3-none-any.whl (3.6 kB)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Downloading tensorboard_data_server-0.6.1-py3-none-any.whl (2.4 kB)
Collecting absl-py>=0.4
  Downloading absl_py-1.3.0-py3-none-any.whl (124 kB)
     |████████████████████████████████| 124 kB 6.4 MB/s
Collecting werkzeug>=1.0.1
  Downloading Werkzeug-2.2.2-py3-none-any.whl (232 kB)
     |████████████████████████████████| 232 kB 3.2 MB/s
Collecting markdown>=2.6.8
  Downloading Markdown-3.4.1-py3-none-any.whl (93 kB)
     |████████████████████████████████| 93 kB 638 kB/s
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Downloading google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting grpcio>=1.24.3
  Downloading grpcio-1.50.0-cp37-cp37m-win_amd64.whl (3.6 MB)
     |████████████████████████████████| 3.6 MB 6.4 MB/s
Collecting wheel>=0.26
  Downloading wheel-0.38.4-py3-none-any.whl (36 kB)
Collecting google-auth<3,>=1.6.3
  Downloading google_auth-2.14.1-py2.py3-none-any.whl (175 kB)
     |████████████████████████████████| 175 kB 6.4 MB/s
Collecting tensorboard-plugin-wit>=1.6.0
  Downloading tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)
     |████████████████████████████████| 781 kB 3.3 MB/s
Collecting sentry-sdk>=1.0.0
  Downloading sentry_sdk-1.11.1-py2.py3-none-any.whl (168 kB)
     |████████████████████████████████| 168 kB 6.4 MB/s
Collecting docker-pycreds>=0.4.0
  Downloading docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Collecting pathtools
  Downloading pathtools-0.1.2.tar.gz (11 kB)
Collecting setproctitle
  Downloading setproctitle-1.3.2-cp37-cp37m-win_amd64.whl (11 kB)
Collecting six>=1.13.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting promise<3,>=2.0
  Downloading promise-2.3.tar.gz (19 kB)
Collecting GitPython>=1.0.0
  Downloading GitPython-3.1.29-py3-none-any.whl (182 kB)
     |████████████████████████████████| 182 kB 6.8 MB/s
Collecting shortuuid>=0.5.0
  Downloading shortuuid-1.0.11-py3-none-any.whl (10 kB)
Collecting mpmath>=0.19
  Downloading mpmath-1.2.1-py3-none-any.whl (532 kB)
     |████████████████████████████████| 532 kB 3.2 MB/s
Collecting humanfriendly>=9.1
  Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
     |████████████████████████████████| 86 kB 1.9 MB/s
Collecting zipp>=0.5
  Downloading zipp-3.11.0-py3-none-any.whl (6.6 kB)
Collecting cffi>=1.12
  Downloading cffi-1.15.1-cp37-cp37m-win_amd64.whl (179 kB)
     |████████████████████████████████| 179 kB 6.4 MB/s
Collecting pystrum>=0.2
  Downloading pystrum-0.2-py3-none-any.whl (19 kB)
Collecting wrapt<2,>=1.10
  Downloading wrapt-1.14.1-cp37-cp37m-win_amd64.whl (35 kB)
Collecting rich<13.0.0,>=10.11.0; extra == "all"
  Downloading rich-12.6.0-py3-none-any.whl (237 kB)
     |████████████████████████████████| 237 kB 6.4 MB/s
Collecting shellingham<2.0.0,>=1.3.0; extra == "all"
  Downloading shellingham-1.5.0-py2.py3-none-any.whl (9.3 kB)
Collecting docopt>=0.6.2
  Using cached docopt-0.6.2.tar.gz (25 kB)
Collecting astor>=0.8
  Downloading astor-0.8.1-py2.py3-none-any.whl (27 kB)
Collecting interface-meta<2.0.0,>=1.2.0
  Using cached interface_meta-1.3.0-py3-none-any.whl (14 kB)
Collecting MarkupSafe>=2.1.1
  Downloading MarkupSafe-2.1.1-cp37-cp37m-win_amd64.whl (17 kB)
Collecting requests-oauthlib>=0.7.0
  Downloading requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting pyasn1-modules>=0.2.1
  Downloading pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
     |████████████████████████████████| 155 kB 6.4 MB/s
Collecting rsa<5,>=3.1.4; python_version >= "3.6"
  Downloading rsa-4.9-py3-none-any.whl (34 kB)
Collecting cachetools<6.0,>=2.0.0
  Downloading cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Collecting gitdb<5,>=4.0.1
  Downloading gitdb-4.0.10-py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 4.8 MB/s
Collecting pyreadline; sys_platform == "win32" and python_version < "3.8"
  Downloading pyreadline-2.1.zip (109 kB)
     |████████████████████████████████| 109 kB 3.3 MB/s
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting commonmark<0.10.0,>=0.9.0
  Downloading commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
     |████████████████████████████████| 51 kB 3.2 MB/s
Collecting pygments<3.0.0,>=2.6.0
  Downloading Pygments-2.13.0-py3-none-any.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 6.4 MB/s
Collecting oauthlib>=3.0.0
  Downloading oauthlib-3.2.2-py3-none-any.whl (151 kB)
     |████████████████████████████████| 151 kB 6.8 MB/s
Collecting pyasn1<0.5.0,>=0.4.6
  Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
     |████████████████████████████████| 77 kB 5.1 MB/s
Collecting smmap<6,>=3.0.1
  Downloading smmap-5.0.0-py3-none-any.whl (24 kB)
Using legacy setup.py install for transforms3d, since package 'wheel' is not installed.
Using legacy setup.py install for pathtools, since package 'wheel' is not installed.
Using legacy setup.py install for promise, since package 'wheel' is not installed.
Using legacy setup.py install for docopt, since package 'wheel' is not installed.
Using legacy setup.py install for pyreadline, since package 'wheel' is not installed.
ERROR: tensorboard 2.11.0 has requirement protobuf<4,>=3.9.2, but you'll have protobuf 4.21.9 which is incompatible.
ERROR: ivadomed 2.9.7 has requirement nibabel~=3.2, but you'll have nibabel 4.0.2 which is incompatible.
Installing collected packages: numpy, pyparsing, packaging, nibabel, h5py, colorama, tqdm, dipy, pillow, typing-extensions, torch, idna, pycparser, cffi, cryptography, pyOpenSSL, urllib3-secure-extra, certifi, urllib3, charset-normalizer, requests, torchvision, scipy, dictdiffer, zipp, importlib-metadata, click, csv-diff, humanize, SimpleITK, wrapt, Deprecated, commonmark, pygments, rich, shellingham, typer, torchio, fonttools, kiwisolver, six, python-dateutil, cycler, matplotlib, imageio, tifffile, networkx, PyWavelets, scikit-image, bids-validator, docopt, num2words, pytz, pandas, astor, interface-meta, formulaic, sqlalchemy, pybids, win32-setctime, loguru, tensorboard-data-server, absl-py, protobuf, MarkupSafe, werkzeug, markdown, oauthlib, requests-oauthlib, pyasn1, pyasn1-modules, rsa, cachetools, google-auth, google-auth-oauthlib, grpcio, wheel, tensorboard-plugin-wit, tensorboard, joblib, threadpoolctl, scikit-learn, sentry-sdk, docker-pycreds, pathtools, psutil, setproctitle, pyyaml, promise, smmap, gitdb, GitPython, shortuuid, wandb, flatbuffers, mpmath, sympy, pyreadline, humanfriendly, coloredlogs, onnxruntime, seaborn, ivadomed, msvc-runtime, lxml, nilearn, pywin32, portalocker, PyQt5-sip, pyqt5, exceptiongroup, tomli, pluggy, attrs, iniconfig, pytest, coverage, pytest-cov, raven, types-setuptools, requirements-parser, xlwt, transforms3d, pytest-console-scripts, pystrum, neurite, voxelmorph, wquantiles
    Running setup.py install for docopt ... done
    Running setup.py install for pathtools ... done
    Running setup.py install for promise ... done
    Running setup.py install for pyreadline ... done
    Running setup.py install for transforms3d ... done
Successfully installed Deprecated-1.2.13 GitPython-3.1.29 MarkupSafe-2.1.1 PyQt5-sip-12.11.0 PyWavelets-1.3.0 SimpleITK-2.2.0 absl-py-1.3.0 astor-0.8.1 attrs-22.1.0 bids-validator-1.9.9 cachetools-5.2.0 certifi-2022.9.24 cffi-1.15.1 charset-normalizer-2.1.1 click-8.1.3 colorama-0.4.6 coloredlogs-15.0.1 commonmark-0.9.1 coverage-6.5.0 cryptography-38.0.4 csv-diff-1.1 cycler-0.11.0 dictdiffer-0.9.0 dipy-1.5.0 docker-pycreds-0.4.0 docopt-0.6.2 exceptiongroup-1.0.4 flatbuffers-22.11.23 fonttools-4.38.0 formulaic-0.3.4 gitdb-4.0.10 google-auth-2.14.1 google-auth-oauthlib-0.4.6 grpcio-1.50.0 h5py-3.7.0 humanfriendly-10.0 humanize-4.4.0 idna-3.4 imageio-2.22.4 importlib-metadata-5.1.0 iniconfig-1.1.1 interface-meta-1.3.0 ivadomed-2.9.7 joblib-1.2.0 kiwisolver-1.4.4 loguru-0.6.0 lxml-4.9.1 markdown-3.4.1 matplotlib-3.5.3 mpmath-1.2.1 msvc-runtime-14.29.30133 networkx-2.6.3 neurite-0.2 nibabel-4.0.2 nilearn-0.9.2 num2words-0.5.12 numpy-1.21.6 oauthlib-3.2.2 onnxruntime-1.13.1 packaging-21.3 pandas-1.3.5 pathtools-0.1.2 pillow-9.3.0 pluggy-1.0.0 portalocker-2.6.0 promise-2.3 protobuf-4.21.9 psutil-5.9.4 pyOpenSSL-22.1.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pybids-0.15.5 pycparser-2.21 pygments-2.13.0 pyparsing-2.4.7 pyqt5-5.14.2 pyreadline-2.1 pystrum-0.2 pytest-7.2.0 pytest-console-scripts-1.3.1 pytest-cov-4.0.0 python-dateutil-2.8.2 pytz-2022.6 pywin32-305 pyyaml-6.0 raven-6.10.0 requests-2.28.1 requests-oauthlib-1.3.1 requirements-parser-0.5.0 rich-12.6.0 rsa-4.9 scikit-image-0.19.3 scikit-learn-1.0.2 scipy-1.7.3 seaborn-0.12.1 sentry-sdk-1.11.1 setproctitle-1.3.2 shellingham-1.5.0 shortuuid-1.0.11 six-1.16.0 smmap-5.0.0 sqlalchemy-1.3.24 sympy-1.10.1 tensorboard-2.11.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 threadpoolctl-3.1.0 tifffile-2021.11.2 tomli-2.0.1 torch-1.11.0 torchio-0.18.86 torchvision-0.12.0 tqdm-4.64.1 transforms3d-0.4.1 typer-0.7.0 types-setuptools-65.6.0.1 typing-extensions-4.4.0 urllib3-1.26.13 urllib3-secure-extra-0.1.0 voxelmorph-0.2 wandb-0.13.5 werkzeug-2.2.2 wheel-0.38.4 win32-setctime-1.1.0 wquantiles-0.6 wrapt-1.14.1 xlwt-1.3.0 zipp-3.11.0
WARNING: You are using pip version 20.1.1; however, version 22.3.1 is available.
You should consider upgrading via the 'c:\users\joshu\repos\spinalcordtoolbox\venv_sct\scripts\python.exe -m pip install --upgrade pip' command.
```
</details>

<details>
<summary> Python 3.7.9, pip 22.3.1 (after running --upgrade) </summary>

```console
venv_sct_upgraded@C:\Users\joshu\repos\spinalcordtoolbox$ python -m pip install --upgrade pip setuptools wheel
Collecting pip
  Downloading pip-22.3.1-py3-none-any.whl (2.1 MB)
     |████████████████████████████████| 2.1 MB 6.8 MB/s
Collecting setuptools
  Downloading setuptools-65.6.3-py3-none-any.whl (1.2 MB)
     |████████████████████████████████| 1.2 MB 6.8 MB/s
Collecting wheel
  Using cached wheel-0.38.4-py3-none-any.whl (36 kB)
Installing collected packages: pip, setuptools, wheel
  Attempting uninstall: pip
    Found existing installation: pip 20.1.1
    Uninstalling pip-20.1.1:
      Successfully uninstalled pip-20.1.1
  Attempting uninstall: setuptools
    Found existing installation: setuptools 47.1.0
    Uninstalling setuptools-47.1.0:
      Successfully uninstalled setuptools-47.1.0
Successfully installed pip-22.3.1 setuptools-65.6.3 wheel-0.38.4

venv_sct_upgraded@C:\Users\joshu\repos\spinalcordtoolbox$ pip install -r requirements.txt
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cpu
Collecting dipy
  Using cached dipy-1.5.0-cp37-none-win_amd64.whl (7.3 MB)
Collecting ivadomed~=2.9.6
  Using cached ivadomed-2.9.7-py3-none-any.whl (231 kB)
Collecting matplotlib
  Using cached matplotlib-3.5.3-cp37-cp37m-win_amd64.whl (7.2 MB)
Collecting msvc-runtime==14.29.30133
  Using cached msvc_runtime-14.29.30133-cp37-cp37m-win_amd64.whl (1.8 MB)
Collecting nibabel
  Using cached nibabel-4.0.2-py3-none-any.whl (3.3 MB)
Collecting nilearn
  Using cached nilearn-0.9.2-py3-none-any.whl (9.6 MB)
Collecting numpy
  Using cached numpy-1.21.6-cp37-cp37m-win_amd64.whl (14.0 MB)
Collecting onnxruntime>=1.7.0
  Using cached onnxruntime-1.13.1-cp37-cp37m-win_amd64.whl (5.9 MB)
Collecting pandas
  Using cached pandas-1.3.5-cp37-cp37m-win_amd64.whl (10.0 MB)
Collecting portalocker
  Using cached portalocker-2.6.0-py2.py3-none-any.whl (15 kB)
Collecting psutil
  Using cached psutil-5.9.4-cp36-abi3-win_amd64.whl (252 kB)
Collecting pyqt5<5.15.0,>=5.12.0
  Using cached PyQt5-5.14.2-5.14.2-cp35.cp36.cp37.cp38-none-win_amd64.whl (52.9 MB)
Collecting pytest
  Using cached pytest-7.2.0-py3-none-any.whl (316 kB)
Collecting pytest-cov
  Using cached pytest_cov-4.0.0-py3-none-any.whl (21 kB)
Collecting raven
  Using cached raven-6.10.0-py2.py3-none-any.whl (284 kB)
Collecting requests
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting requirements-parser
  Using cached requirements_parser-0.5.0-py3-none-any.whl (18 kB)
Collecting scipy
  Using cached scipy-1.7.3-cp37-cp37m-win_amd64.whl (34.1 MB)
Collecting scikit-image
  Using cached scikit_image-0.19.3-cp37-cp37m-win_amd64.whl (12.1 MB)
Collecting scikit-learn
  Using cached scikit_learn-1.0.2-cp37-cp37m-win_amd64.whl (7.1 MB)
Collecting xlwt
  Using cached xlwt-1.3.0-py2.py3-none-any.whl (99 kB)
Collecting tqdm
  Using cached tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
Collecting transforms3d
  Using cached transforms3d-0.4.1.tar.gz (1.4 MB)
  Preparing metadata (setup.py) ... done
Collecting urllib3[secure]
  Using cached urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
Collecting pytest_console_scripts
  Using cached pytest_console_scripts-1.3.1-py3-none-any.whl (9.0 kB)
Collecting pyyaml
  Using cached PyYAML-6.0-cp37-cp37m-win_amd64.whl (153 kB)
Collecting voxelmorph
  Using cached voxelmorph-0.2-py3-none-any.whl (54 kB)
Collecting wquantiles
  Using cached wquantiles-0.6-py3-none-any.whl (3.3 kB)
Collecting h5py>=2.8.0
  Using cached h5py-3.7.0-cp37-cp37m-win_amd64.whl (2.6 MB)
Collecting torchvision<=0.12.0,>=0.9.1
  Downloading https://download.pytorch.org/whl/cpu/torchvision-0.12.0%2Bcpu-cp37-cp37m-win_amd64.whl (1.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.0/1.0 MB 10.6 MB/s eta 0:00:00
Collecting imageio~=2.19
  Using cached imageio-2.22.4-py3-none-any.whl (3.4 MB)
Collecting joblib~=1.0
  Using cached joblib-1.2.0-py3-none-any.whl (297 kB)
Collecting tensorboard>=1.15.0
  Using cached tensorboard-2.11.0-py3-none-any.whl (6.0 MB)
Collecting seaborn~=0.11
  Using cached seaborn-0.12.1-py3-none-any.whl (288 kB)
Collecting loguru~=0.5
  Using cached loguru-0.6.0-py3-none-any.whl (58 kB)
Collecting nibabel
  Using cached nibabel-3.2.2-py3-none-any.whl (3.3 MB)
Collecting pybids>=0.14.0
  Using cached pybids-0.15.5-py3-none-any.whl (14.9 MB)
Collecting wandb>=0.12.11
  Using cached wandb-0.13.5-py2.py3-none-any.whl (1.9 MB)
Collecting csv-diff>=1.0
  Using cached csv_diff-1.1-py3-none-any.whl (12 kB)
Collecting torch<=1.11.0,>=1.8.1
  Downloading https://download.pytorch.org/whl/cpu/torch-1.11.0%2Bcpu-cp37-cp37m-win_amd64.whl (157.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 157.5/157.5 MB 3.5 MB/s eta 0:00:00
Collecting pyparsing<3,>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting torchio>=0.18.68
  Using cached torchio-0.18.86-py2.py3-none-any.whl (172 kB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.38.0-py3-none-any.whl (965 kB)
Collecting pillow>=6.2.0
  Using cached Pillow-9.3.0-1-cp37-cp37m-win_amd64.whl (2.5 MB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.4.4-cp37-cp37m-win_amd64.whl (54 kB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Requirement already satisfied: setuptools in c:\users\joshu\repos\spinalcordtoolbox\venv_sct_upgraded\lib\site-packages (from nibabel->-r requirements.txt (line 18)) (65.6.3)
Collecting lxml
  Using cached lxml-4.9.1-cp37-cp37m-win_amd64.whl (3.6 MB)
Collecting protobuf
  Using cached protobuf-4.21.9-cp37-cp37m-win_amd64.whl (525 kB)
Collecting sympy
  Using cached sympy-1.10.1-py3-none-any.whl (6.4 MB)
Collecting coloredlogs
  Using cached coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
Collecting flatbuffers
  Using cached flatbuffers-22.11.23-py2.py3-none-any.whl (26 kB)
Collecting pytz>=2017.3
  Using cached pytz-2022.6-py2.py3-none-any.whl (498 kB)
Collecting pywin32>=226
  Using cached pywin32-305-cp37-cp37m-win_amd64.whl (12.2 MB)
Collecting PyQt5-sip<13,>=12.7
  Using cached PyQt5_sip-12.11.0-cp37-cp37m-win_amd64.whl (77 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting tomli>=1.0.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting attrs>=19.2.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting importlib-metadata>=0.12
  Using cached importlib_metadata-5.1.0-py3-none-any.whl (21 kB)
Collecting exceptiongroup>=1.0.0rc8
  Using cached exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
Collecting coverage[toml]>=5.2.1
  Using cached coverage-6.5.0-cp37-cp37m-win_amd64.whl (188 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting types-setuptools>=57.0.0
  Using cached types_setuptools-65.6.0.1-py3-none-any.whl (48 kB)
Collecting networkx>=2.2
  Using cached networkx-2.6.3-py3-none-any.whl (1.9 MB)
Collecting tifffile>=2019.7.26
  Using cached tifffile-2021.11.2-py3-none-any.whl (178 kB)
Collecting PyWavelets>=1.1.1
  Using cached PyWavelets-1.3.0-cp37-cp37m-win_amd64.whl (4.2 MB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting cryptography>=1.3.4
  Using cached cryptography-38.0.4-cp36-abi3-win_amd64.whl (2.4 MB)
Collecting pyOpenSSL>=0.14
  Using cached pyOpenSSL-22.1.0-py3-none-any.whl (57 kB)
Collecting urllib3-secure-extra
  Using cached urllib3_secure_extra-0.1.0-py2.py3-none-any.whl (1.4 kB)
Collecting neurite>=0.2
  Using cached neurite-0.2-py3-none-any.whl (108 kB)
Collecting cffi>=1.12
  Using cached cffi-1.15.1-cp37-cp37m-win_amd64.whl (179 kB)
Collecting click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting dictdiffer
  Using cached dictdiffer-0.9.0-py2.py3-none-any.whl (16 kB)
Collecting typing-extensions>=3.6.4
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting zipp>=0.5
  Using cached zipp-3.11.0-py3-none-any.whl (6.6 kB)
Collecting win32-setctime>=1.0.0
  Using cached win32_setctime-1.1.0-py3-none-any.whl (3.6 kB)
Collecting pystrum>=0.2
  Using cached pystrum-0.2-py3-none-any.whl (19 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting formulaic<0.4,>=0.2.4
  Using cached formulaic-0.3.4-py3-none-any.whl (68 kB)
Collecting sqlalchemy<1.4.0.dev0
  Using cached SQLAlchemy-1.3.24-cp37-cp37m-win_amd64.whl (1.2 MB)
Collecting bids-validator
  Using cached bids_validator-1.9.9-py2.py3-none-any.whl (21 kB)
Collecting num2words
  Using cached num2words-0.5.12-py3-none-any.whl (125 kB)
Collecting markdown>=2.6.8
  Using cached Markdown-3.4.1-py3-none-any.whl (93 kB)
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)
Collecting absl-py>=0.4
  Using cached absl_py-1.3.0-py3-none-any.whl (124 kB)
Collecting google-auth<3,>=1.6.3
  Using cached google_auth-2.14.1-py2.py3-none-any.whl (175 kB)
Collecting werkzeug>=1.0.1
  Using cached Werkzeug-2.2.2-py3-none-any.whl (232 kB)
Requirement already satisfied: wheel>=0.26 in c:\users\joshu\repos\spinalcordtoolbox\venv_sct_upgraded\lib\site-packages (from tensorboard>=1.15.0->ivadomed~=2.9.6->-r requirements.txt (line 9)) (0.38.4)
Collecting grpcio>=1.24.3
  Using cached grpcio-1.50.0-cp37-cp37m-win_amd64.whl (3.6 MB)
Collecting protobuf
  Downloading protobuf-3.20.3-cp37-cp37m-win_amd64.whl (905 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 905.1/905.1 kB 14.2 MB/s eta 0:00:00
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-any.whl (2.4 kB)
Collecting SimpleITK!=2.0.*,!=2.1.1.1
  Using cached SimpleITK-2.2.0-cp37-cp37m-win_amd64.whl (27.1 MB)
Collecting Deprecated
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting typer[all]
  Using cached typer-0.7.0-py3-none-any.whl (38 kB)
Collecting humanize
  Using cached humanize-4.4.0-py3-none-any.whl (106 kB)
Collecting GitPython>=1.0.0
  Using cached GitPython-3.1.29-py3-none-any.whl (182 kB)
Collecting setproctitle
  Using cached setproctitle-1.3.2-cp37-cp37m-win_amd64.whl (11 kB)
Collecting docker-pycreds>=0.4.0
  Using cached docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Collecting promise<3,>=2.0
  Using cached promise-2.3.tar.gz (19 kB)
  Preparing metadata (setup.py) ... done
Collecting pathtools
  Using cached pathtools-0.1.2.tar.gz (11 kB)
  Preparing metadata (setup.py) ... done
Collecting sentry-sdk>=1.0.0
  Using cached sentry_sdk-1.11.1-py2.py3-none-any.whl (168 kB)
Collecting shortuuid>=0.5.0
  Using cached shortuuid-1.0.11-py3-none-any.whl (10 kB)
Collecting humanfriendly>=9.1
  Using cached humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
Collecting mpmath>=0.19
  Using cached mpmath-1.2.1-py3-none-any.whl (532 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting interface-meta<2.0.0,>=1.2.0
  Using cached interface_meta-1.3.0-py3-none-any.whl (14 kB)
Collecting astor>=0.8
  Using cached astor-0.8.1-py2.py3-none-any.whl (27 kB)
Collecting wrapt>=1.0
  Using cached wrapt-1.14.1-cp37-cp37m-win_amd64.whl (35 kB)
Collecting gitdb<5,>=4.0.1
  Using cached gitdb-4.0.10-py3-none-any.whl (62 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting pyreadline
  Using cached pyreadline-2.1.zip (109 kB)
  Preparing metadata (setup.py) ... done
Collecting MarkupSafe>=2.1.1
  Using cached MarkupSafe-2.1.1-cp37-cp37m-win_amd64.whl (17 kB)
Collecting docopt>=0.6.2
  Using cached docopt-0.6.2.tar.gz (25 kB)
  Preparing metadata (setup.py) ... done
Collecting shellingham<2.0.0,>=1.3.0
  Using cached shellingham-1.5.0-py2.py3-none-any.whl (9.3 kB)
Collecting rich<13.0.0,>=10.11.0
  Using cached rich-12.6.0-py3-none-any.whl (237 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Collecting commonmark<0.10.0,>=0.9.0
  Using cached commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
Collecting pygments<3.0.0,>=2.6.0
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Building wheels for collected packages: transforms3d, promise, pathtools, docopt, pyreadline
  Building wheel for transforms3d (setup.py) ... done
  Created wheel for transforms3d: filename=transforms3d-0.4.1-py3-none-any.whl size=1376797 sha256=9b95b34cef21c7e39971ebec40939ca4b42f866c58702035186b68b803c97b4f
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\0b\2a\fe\379cf61e108f06750107439af6531daf372d7de2c390a939b4
  Building wheel for promise (setup.py) ... done
  Created wheel for promise: filename=promise-2.3-py3-none-any.whl size=21548 sha256=f9b08430b6e8d9fb14ffc1e4451a7ac4be2a4d2a9b1afae7c0b2d7f68586391b
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\9d\ad\15\e6d5c43a0f01b88ee5883bd249a18e09d72821e43b1c3e8187
  Building wheel for pathtools (setup.py) ... done
  Created wheel for pathtools: filename=pathtools-0.1.2-py3-none-any.whl size=8801 sha256=52dbbb905c46816d94dcece81ccb456d0ce30bde9c9b1fa96c1ddeeda68c7435
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\7b\d8\86\08d5794e37091c7a918e0bb3322b7b43af55bda3b665b81e05
  Building wheel for docopt (setup.py) ... done
  Created wheel for docopt: filename=docopt-0.6.2-py2.py3-none-any.whl size=13775 sha256=d8440d5d88bbad5e0b62b01c2702d0e5422c73fa363a260362559cc03ac1a4d6
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\1f\e0\34\704e15d0b49f199ac74a2757d018d428ad4cf119c078c80f16
  Building wheel for pyreadline (setup.py) ... done
  Created wheel for pyreadline: filename=pyreadline-2.1-py3-none-any.whl size=93855 sha256=cc1a660b34dbd49372cc56b56db8b05e79e959971dbd9c250cb7326bef1107ae
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\2c\ba\0a\79cfdda2594b992540020a641a4abc70764064f532acbe56b4
Successfully built transforms3d promise pathtools docopt pyreadline
Installing collected packages: xlwt, urllib3-secure-extra, types-setuptools, tensorboard-plugin-wit, SimpleITK, raven, pywin32, pytz, pyreadline, pyasn1, pathtools, mpmath, iniconfig, flatbuffers, docopt, dictdiffer, commonmark, bids-validator, zipp, wrapt, win32-setctime, urllib3, typing-extensions, transforms3d, tomli, threadpoolctl, tensorboard-data-server, sympy, sqlalchemy, smmap, six, shortuuid, shellingham, setproctitle, rsa, requirements-parser, pyyaml, PyQt5-sip, pyparsing, pygments, pycparser, pyasn1-modules, psutil, protobuf, portalocker, pillow, oauthlib, numpy, num2words, networkx, msvc-runtime, MarkupSafe, lxml, joblib, interface-meta, idna, humanfriendly, fonttools, exceptiongroup, cycler, coverage, colorama, charset-normalizer, certifi, cachetools, attrs, astor, absl-py, wquantiles, werkzeug, tqdm, torch, tifffile, sentry-sdk, scipy, rich, requests, PyWavelets, python-dateutil, pyqt5, promise, packaging, loguru, kiwisolver, importlib-metadata, imageio, h5py, grpcio, google-auth, gitdb, docker-pycreds, Deprecated, coloredlogs, cffi, torchvision, scikit-learn, scikit-image, requests-oauthlib, pluggy, pandas, onnxruntime, nibabel, matplotlib, markdown, humanize, GitPython, cryptography, click, wandb, typer, seaborn, pytest, pystrum, pyOpenSSL, nilearn, google-auth-oauthlib, formulaic, dipy, csv-diff, tensorboard, pytest-cov, pytest_console_scripts, pybids, neurite, voxelmorph, torchio, ivadomed
Successfully installed Deprecated-1.2.13 GitPython-3.1.29 MarkupSafe-2.1.1 PyQt5-sip-12.11.0 PyWavelets-1.3.0 SimpleITK-2.2.0 absl-py-1.3.0 astor-0.8.1 attrs-22.1.0 bids-validator-1.9.9 cachetools-5.2.0 certifi-2022.9.24 cffi-1.15.1 charset-normalizer-2.1.1 click-8.1.3 colorama-0.4.6 coloredlogs-15.0.1 commonmark-0.9.1 coverage-6.5.0 cryptography-38.0.4 csv-diff-1.1 cycler-0.11.0 dictdiffer-0.9.0 dipy-1.5.0 docker-pycreds-0.4.0 docopt-0.6.2 exceptiongroup-1.0.4 flatbuffers-22.11.23 fonttools-4.38.0 formulaic-0.3.4 gitdb-4.0.10 google-auth-2.14.1 google-auth-oauthlib-0.4.6 grpcio-1.50.0 h5py-3.7.0 humanfriendly-10.0 humanize-4.4.0 idna-3.4 imageio-2.22.4 importlib-metadata-5.1.0 iniconfig-1.1.1 interface-meta-1.3.0 ivadomed-2.9.7 joblib-1.2.0 kiwisolver-1.4.4 loguru-0.6.0 lxml-4.9.1 markdown-3.4.1 matplotlib-3.5.3 mpmath-1.2.1 msvc-runtime-14.29.30133 networkx-2.6.3 neurite-0.2 nibabel-3.2.2 nilearn-0.9.2 num2words-0.5.12 numpy-1.21.6 oauthlib-3.2.2 onnxruntime-1.13.1 packaging-21.3 pandas-1.3.5 pathtools-0.1.2 pillow-9.3.0 pluggy-1.0.0 portalocker-2.6.0 promise-2.3 protobuf-3.20.3 psutil-5.9.4 pyOpenSSL-22.1.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pybids-0.15.5 pycparser-2.21 pygments-2.13.0 pyparsing-2.4.7 pyqt5-5.14.2 pyreadline-2.1 pystrum-0.2 pytest-7.2.0 pytest-cov-4.0.0 pytest_console_scripts-1.3.1 python-dateutil-2.8.2 pytz-2022.6 pywin32-305 pyyaml-6.0 raven-6.10.0 requests-2.28.1 requests-oauthlib-1.3.1 requirements-parser-0.5.0 rich-12.6.0 rsa-4.9 scikit-image-0.19.3 scikit-learn-1.0.2 scipy-1.7.3 seaborn-0.12.1 sentry-sdk-1.11.1 setproctitle-1.3.2 shellingham-1.5.0 shortuuid-1.0.11 six-1.16.0 smmap-5.0.0 sqlalchemy-1.3.24 sympy-1.10.1 tensorboard-2.11.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 threadpoolctl-3.1.0 tifffile-2021.11.2 tomli-2.0.1 torch-1.11.0+cpu torchio-0.18.86 torchvision-0.12.0+cpu tqdm-4.64.1 transforms3d-0.4.1 typer-0.7.0 types-setuptools-65.6.0.1 typing-extensions-4.4.0 urllib3-1.26.13 urllib3-secure-extra-0.1.0 voxelmorph-0.2 wandb-0.13.5 werkzeug-2.2.2 win32-setctime-1.1.0 wquantiles-0.6 wrapt-1.14.1 xlwt-1.3.0 zipp-3.11.0
```
</details>

<details>
<summary> Python 3.8.10, pip 21.1.1 (no upgrade) </summary>

```console
venv_sct_38@C:\Users\joshu\repos\spinalcordtoolbox$ pip install -r requirements.txt
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cpu
Collecting dipy
  Downloading dipy-1.5.0-cp38-none-win_amd64.whl (7.4 MB)
     |████████████████████████████████| 7.4 MB 6.4 MB/s
Collecting ivadomed~=2.9.6
  Using cached ivadomed-2.9.7-py3-none-any.whl (231 kB)
Collecting matplotlib
  Downloading matplotlib-3.6.2-cp38-cp38-win_amd64.whl (7.2 MB)
     |████████████████████████████████| 7.2 MB 6.8 MB/s
Collecting msvc-runtime==14.29.30133
  Downloading msvc_runtime-14.29.30133-cp38-cp38-win_amd64.whl (1.8 MB)
     |████████████████████████████████| 1.8 MB 6.8 MB/s
Collecting nibabel
  Using cached nibabel-4.0.2-py3-none-any.whl (3.3 MB)
Collecting nilearn
  Using cached nilearn-0.9.2-py3-none-any.whl (9.6 MB)
Collecting numpy
  Downloading numpy-1.23.5-cp38-cp38-win_amd64.whl (14.7 MB)
     |████████████████████████████████| 14.7 MB 6.8 MB/s
Collecting onnxruntime>=1.7.0
  Downloading onnxruntime-1.13.1-cp38-cp38-win_amd64.whl (5.9 MB)
     |████████████████████████████████| 5.9 MB 6.4 MB/s
Collecting pandas
  Downloading pandas-1.5.2-cp38-cp38-win_amd64.whl (11.0 MB)
     |████████████████████████████████| 11.0 MB 6.8 MB/s
Collecting portalocker
  Using cached portalocker-2.6.0-py2.py3-none-any.whl (15 kB)
Collecting psutil
  Using cached psutil-5.9.4-cp36-abi3-win_amd64.whl (252 kB)
Collecting pyqt5<5.15.0,>=5.12.0
  Using cached PyQt5-5.14.2-5.14.2-cp35.cp36.cp37.cp38-none-win_amd64.whl (52.9 MB)
Collecting pytest
  Using cached pytest-7.2.0-py3-none-any.whl (316 kB)
Collecting pytest-cov
  Using cached pytest_cov-4.0.0-py3-none-any.whl (21 kB)
Collecting raven
  Using cached raven-6.10.0-py2.py3-none-any.whl (284 kB)
Collecting requests
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting requirements-parser
  Using cached requirements_parser-0.5.0-py3-none-any.whl (18 kB)
Collecting scipy
  Downloading scipy-1.9.3-cp38-cp38-win_amd64.whl (39.8 MB)
     |████████████████████████████████| 39.8 MB 238 kB/s
Collecting scikit-image
  Downloading scikit_image-0.19.3-cp38-cp38-win_amd64.whl (12.2 MB)
     |████████████████████████████████| 12.2 MB 6.4 MB/s
Collecting scikit-learn
  Downloading scikit_learn-1.1.3-cp38-cp38-win_amd64.whl (7.5 MB)
     |████████████████████████████████| 7.5 MB 6.8 MB/s
Collecting xlwt
  Using cached xlwt-1.3.0-py2.py3-none-any.whl (99 kB)
Collecting tqdm
  Using cached tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
Collecting transforms3d
  Using cached transforms3d-0.4.1.tar.gz (1.4 MB)
Collecting urllib3[secure]
  Using cached urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
Collecting pytest_console_scripts
  Using cached pytest_console_scripts-1.3.1-py3-none-any.whl (9.0 kB)
Collecting pyyaml
  Downloading PyYAML-6.0-cp38-cp38-win_amd64.whl (155 kB)
     |████████████████████████████████| 155 kB 6.4 MB/s
Collecting voxelmorph
  Using cached voxelmorph-0.2-py3-none-any.whl (54 kB)
Collecting wquantiles
  Using cached wquantiles-0.6-py3-none-any.whl (3.3 kB)
Collecting seaborn~=0.11
  Using cached seaborn-0.12.1-py3-none-any.whl (288 kB)
Collecting pybids>=0.14.0
  Using cached pybids-0.15.5-py3-none-any.whl (14.9 MB)
Collecting pyparsing<3,>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting nibabel
  Using cached nibabel-3.2.2-py3-none-any.whl (3.3 MB)
Collecting torchvision<=0.12.0,>=0.9.1
  Downloading https://download.pytorch.org/whl/cpu/torchvision-0.12.0%2Bcpu-cp38-cp38-win_amd64.whl (1.0 MB)
     |████████████████████████████████| 1.0 MB ...
Collecting joblib~=1.0
  Using cached joblib-1.2.0-py3-none-any.whl (297 kB)
Collecting imageio~=2.19
  Using cached imageio-2.22.4-py3-none-any.whl (3.4 MB)
Collecting wandb>=0.12.11
  Using cached wandb-0.13.5-py2.py3-none-any.whl (1.9 MB)
Collecting tensorboard>=1.15.0
  Using cached tensorboard-2.11.0-py3-none-any.whl (6.0 MB)
Collecting torchio>=0.18.68
  Using cached torchio-0.18.86-py2.py3-none-any.whl (172 kB)
Collecting torch<=1.11.0,>=1.8.1
  Downloading https://download.pytorch.org/whl/cpu/torch-1.11.0%2Bcpu-cp38-cp38-win_amd64.whl (157.5 MB)
     |████████████████████████████████| 157.5 MB 102 kB/s
Collecting loguru~=0.5
  Using cached loguru-0.6.0-py3-none-any.whl (58 kB)
Collecting csv-diff>=1.0
  Using cached csv_diff-1.1-py3-none-any.whl (12 kB)
Collecting pandas
  Downloading pandas-1.4.4-cp38-cp38-win_amd64.whl (10.6 MB)
     |████████████████████████████████| 10.6 MB 6.4 MB/s
Collecting pillow>=6.2.0
  Downloading Pillow-9.3.0-cp38-cp38-win_amd64.whl (2.5 MB)
     |████████████████████████████████| 2.5 MB 6.4 MB/s
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.38.0-py3-none-any.whl (965 kB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting kiwisolver>=1.0.1
  Downloading kiwisolver-1.4.4-cp38-cp38-win_amd64.whl (55 kB)
     |████████████████████████████████| 55 kB 3.8 MB/s
Collecting contourpy>=1.0.1
  Downloading contourpy-1.0.6-cp38-cp38-win_amd64.whl (163 kB)
     |████████████████████████████████| 163 kB ...
Requirement already satisfied: setuptools in c:\users\joshu\repos\spinalcordtoolbox\venv_sct_38\lib\site-packages (from nibabel->-r requirements.txt (line 18)) (56.0.0)
Collecting protobuf
  Downloading protobuf-4.21.9-cp38-cp38-win_amd64.whl (525 kB)
     |████████████████████████████████| 525 kB 6.8 MB/s
Collecting sympy
  Downloading sympy-1.11.1-py3-none-any.whl (6.5 MB)
     |████████████████████████████████| 6.5 MB ...
Collecting coloredlogs
  Using cached coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
Collecting flatbuffers
  Using cached flatbuffers-22.11.23-py2.py3-none-any.whl (26 kB)
Collecting pytz>=2020.1
  Using cached pytz-2022.6-py2.py3-none-any.whl (498 kB)
Collecting PyQt5-sip<13,>=12.7
  Downloading PyQt5_sip-12.11.0-cp38-cp38-win_amd64.whl (78 kB)
     |████████████████████████████████| 78 kB 3.2 MB/s
Collecting tifffile>=2019.7.26
  Downloading tifffile-2022.10.10-py3-none-any.whl (210 kB)
     |████████████████████████████████| 210 kB 6.8 MB/s
Collecting networkx>=2.2
  Downloading networkx-2.8.8-py3-none-any.whl (2.0 MB)
     |████████████████████████████████| 2.0 MB 6.4 MB/s
Collecting PyWavelets>=1.1.1
  Downloading PyWavelets-1.4.1-cp38-cp38-win_amd64.whl (4.2 MB)
     |████████████████████████████████| 4.2 MB 6.8 MB/s
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting dictdiffer
  Using cached dictdiffer-0.9.0-py2.py3-none-any.whl (16 kB)
Collecting win32-setctime>=1.0.0
  Using cached win32_setctime-1.1.0-py3-none-any.whl (3.6 kB)
Collecting num2words
  Using cached num2words-0.5.12-py3-none-any.whl (125 kB)
Collecting sqlalchemy<1.4.0.dev0
  Downloading SQLAlchemy-1.3.24-cp38-cp38-win_amd64.whl (1.2 MB)
     |████████████████████████████████| 1.2 MB 6.4 MB/s
Collecting bids-validator
  Using cached bids_validator-1.9.9-py2.py3-none-any.whl (21 kB)
Collecting formulaic<0.4,>=0.2.4
  Using cached formulaic-0.3.4-py3-none-any.whl (68 kB)
Collecting astor>=0.8
  Using cached astor-0.8.1-py2.py3-none-any.whl (27 kB)
Collecting wrapt>=1.0
  Downloading wrapt-1.14.1-cp38-cp38-win_amd64.whl (35 kB)
Collecting interface-meta<2.0.0,>=1.2.0
  Using cached interface_meta-1.3.0-py3-none-any.whl (14 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting protobuf
  Downloading protobuf-3.20.3-cp38-cp38-win_amd64.whl (904 kB)
     |████████████████████████████████| 904 kB 6.4 MB/s
Collecting markdown>=2.6.8
  Using cached Markdown-3.4.1-py3-none-any.whl (93 kB)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-any.whl (2.4 kB)
Collecting google-auth<3,>=1.6.3
  Using cached google_auth-2.14.1-py2.py3-none-any.whl (175 kB)
Collecting werkzeug>=1.0.1
  Using cached Werkzeug-2.2.2-py3-none-any.whl (232 kB)
Collecting grpcio>=1.24.3
  Downloading grpcio-1.50.0-cp38-cp38-win_amd64.whl (3.7 MB)
     |████████████████████████████████| 3.7 MB 6.4 MB/s
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)
Collecting wheel>=0.26
  Using cached wheel-0.38.4-py3-none-any.whl (36 kB)
Collecting absl-py>=0.4
  Using cached absl_py-1.3.0-py3-none-any.whl (124 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting importlib-metadata>=4.4
  Using cached importlib_metadata-5.1.0-py3-none-any.whl (21 kB)
Collecting zipp>=0.5
  Using cached zipp-3.11.0-py3-none-any.whl (6.6 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Collecting typing-extensions
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting humanize
  Using cached humanize-4.4.0-py3-none-any.whl (106 kB)
Collecting SimpleITK!=2.0.*,!=2.1.1.1
  Downloading SimpleITK-2.2.0-cp38-cp38-win_amd64.whl (27.1 MB)
     |████████████████████████████████| 27.1 MB 6.4 MB/s
Collecting Deprecated
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting typer[all]
  Using cached typer-0.7.0-py3-none-any.whl (38 kB)
Collecting pathtools
  Using cached pathtools-0.1.2.tar.gz (11 kB)
Collecting GitPython>=1.0.0
  Using cached GitPython-3.1.29-py3-none-any.whl (182 kB)
Collecting shortuuid>=0.5.0
  Using cached shortuuid-1.0.11-py3-none-any.whl (10 kB)
Collecting setproctitle
  Downloading setproctitle-1.3.2-cp38-cp38-win_amd64.whl (11 kB)
Collecting sentry-sdk>=1.0.0
  Using cached sentry_sdk-1.11.1-py2.py3-none-any.whl (168 kB)
Collecting docker-pycreds>=0.4.0
  Using cached docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Collecting promise<3,>=2.0
  Using cached promise-2.3.tar.gz (19 kB)
Collecting gitdb<5,>=4.0.1
  Using cached gitdb-4.0.10-py3-none-any.whl (62 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting MarkupSafe>=2.1.1
  Downloading MarkupSafe-2.1.1-cp38-cp38-win_amd64.whl (17 kB)
Collecting h5py>=2.8.0
  Using cached h5py-3.7.0-cp38-cp38-win_amd64.whl (2.6 MB)
Collecting lxml
  Downloading lxml-4.9.1-cp38-cp38-win_amd64.whl (3.6 MB)
     |████████████████████████████████| 3.6 MB 6.4 MB/s
Collecting pywin32>=226
  Downloading pywin32-305-cp38-cp38-win_amd64.whl (12.3 MB)
     |████████████████████████████████| 12.3 MB 6.8 MB/s
Collecting tomli>=1.0.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting attrs>=19.2.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting exceptiongroup>=1.0.0rc8
  Using cached exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
Collecting coverage[toml]>=5.2.1
  Downloading coverage-6.5.0-cp38-cp38-win_amd64.whl (188 kB)
     |████████████████████████████████| 188 kB 6.8 MB/s
Collecting types-setuptools>=57.0.0
  Using cached types_setuptools-65.6.0.1-py3-none-any.whl (48 kB)
Collecting neurite>=0.2
  Using cached neurite-0.2-py3-none-any.whl (108 kB)
Collecting pystrum>=0.2
  Using cached pystrum-0.2-py3-none-any.whl (19 kB)
Collecting humanfriendly>=9.1
  Using cached humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
Collecting pyreadline3
  Downloading pyreadline3-3.4.1-py3-none-any.whl (95 kB)
     |████████████████████████████████| 95 kB 6.8 MB/s
Collecting docopt>=0.6.2
  Using cached docopt-0.6.2.tar.gz (25 kB)
Collecting mpmath>=0.19
  Using cached mpmath-1.2.1-py3-none-any.whl (532 kB)
Collecting shellingham<2.0.0,>=1.3.0
  Using cached shellingham-1.5.0-py2.py3-none-any.whl (9.3 kB)
Collecting rich<13.0.0,>=10.11.0
  Using cached rich-12.6.0-py3-none-any.whl (237 kB)
Collecting pygments<3.0.0,>=2.6.0
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Collecting commonmark<0.10.0,>=0.9.0
  Using cached commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
Collecting pyOpenSSL>=0.14
  Using cached pyOpenSSL-22.1.0-py3-none-any.whl (57 kB)
Collecting urllib3-secure-extra
  Using cached urllib3_secure_extra-0.1.0-py2.py3-none-any.whl (1.4 kB)
Collecting cryptography>=1.3.4
  Using cached cryptography-38.0.4-cp36-abi3-win_amd64.whl (2.4 MB)
Collecting cffi>=1.12
  Downloading cffi-1.15.1-cp38-cp38-win_amd64.whl (178 kB)
     |████████████████████████████████| 178 kB 6.8 MB/s
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Using legacy 'setup.py install' for promise, since package 'wheel' is not installed.
Using legacy 'setup.py install' for transforms3d, since package 'wheel' is not installed.
Using legacy 'setup.py install' for docopt, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pathtools, since package 'wheel' is not installed.
Installing collected packages: urllib3, six, pyparsing, pyasn1, numpy, idna, colorama, charset-normalizer, certifi, zipp, typing-extensions, smmap, rsa, requests, pytz, python-dateutil, pyreadline3, pygments, pycparser, pyasn1-modules, pillow, packaging, oauthlib, kiwisolver, fonttools, cycler, contourpy, commonmark, click, cachetools, wrapt, typer, threadpoolctl, shellingham, scipy, rich, requests-oauthlib, pandas, mpmath, matplotlib, MarkupSafe, joblib, interface-meta, importlib-metadata, humanfriendly, google-auth, gitdb, docopt, cffi, astor, win32-setctime, wheel, werkzeug, tqdm, torch, tomli, tifffile, tensorboard-plugin-wit, tensorboard-data-server, sympy, sqlalchemy, SimpleITK, shortuuid, setproctitle, sentry-sdk, scikit-learn, pyyaml, PyWavelets, pystrum, psutil, protobuf, promise, pluggy, pathtools, num2words, nibabel, networkx, markdown, iniconfig, imageio, humanize, grpcio, google-auth-oauthlib, GitPython, formulaic, flatbuffers, exceptiongroup, docker-pycreds, dictdiffer, Deprecated, cryptography, coverage, coloredlogs, bids-validator, attrs, absl-py, wandb, urllib3-secure-extra, types-setuptools, torchvision, torchio, tensorboard, seaborn, scikit-image, pywin32, pytest, PyQt5-sip, pyOpenSSL, pybids, onnxruntime, neurite, lxml, loguru, h5py, csv-diff, xlwt, wquantiles, voxelmorph, transforms3d, requirements-parser, raven, pytest-cov, pytest-console-scripts, pyqt5, portalocker, nilearn, msvc-runtime, ivadomed, dipy
    Running setup.py install for docopt ... done
    Running setup.py install for promise ... done
    Running setup.py install for pathtools ... done
    Running setup.py install for transforms3d ... done
Successfully installed Deprecated-1.2.13 GitPython-3.1.29 MarkupSafe-2.1.1 PyQt5-sip-12.11.0 PyWavelets-1.4.1 SimpleITK-2.2.0 absl-py-1.3.0 astor-0.8.1 attrs-22.1.0 bids-validator-1.9.9 cachetools-5.2.0 certifi-2022.9.24 cffi-1.15.1 charset-normalizer-2.1.1 click-8.1.3 colorama-0.4.6 coloredlogs-15.0.1 commonmark-0.9.1 contourpy-1.0.6 coverage-6.5.0 cryptography-38.0.4 csv-diff-1.1 cycler-0.11.0 dictdiffer-0.9.0 dipy-1.5.0 docker-pycreds-0.4.0 docopt-0.6.2 exceptiongroup-1.0.4 flatbuffers-22.11.23 fonttools-4.38.0 formulaic-0.3.4 gitdb-4.0.10 google-auth-2.14.1 google-auth-oauthlib-0.4.6 grpcio-1.50.0 h5py-3.7.0 humanfriendly-10.0 humanize-4.4.0 idna-3.4 imageio-2.22.4 importlib-metadata-5.1.0 iniconfig-1.1.1 interface-meta-1.3.0 ivadomed-2.9.7 joblib-1.2.0 kiwisolver-1.4.4 loguru-0.6.0 lxml-4.9.1 markdown-3.4.1 matplotlib-3.6.2 mpmath-1.2.1 msvc-runtime-14.29.30133 networkx-2.8.8 neurite-0.2 nibabel-3.2.2 nilearn-0.9.2 num2words-0.5.12 numpy-1.23.5 oauthlib-3.2.2 onnxruntime-1.13.1 packaging-21.3 pandas-1.4.4 pathtools-0.1.2 pillow-9.3.0 pluggy-1.0.0 portalocker-2.6.0 promise-2.3 protobuf-3.20.3 psutil-5.9.4 pyOpenSSL-22.1.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pybids-0.15.5 pycparser-2.21 pygments-2.13.0 pyparsing-2.4.7 pyqt5-5.14.2 pyreadline3-3.4.1 pystrum-0.2 pytest-7.2.0 pytest-console-scripts-1.3.1 pytest-cov-4.0.0 python-dateutil-2.8.2 pytz-2022.6 pywin32-305 pyyaml-6.0 raven-6.10.0 requests-2.28.1 requests-oauthlib-1.3.1 requirements-parser-0.5.0 rich-12.6.0 rsa-4.9 scikit-image-0.19.3 scikit-learn-1.1.3 scipy-1.9.3 seaborn-0.12.1 sentry-sdk-1.11.1 setproctitle-1.3.2 shellingham-1.5.0 shortuuid-1.0.11 six-1.16.0 smmap-5.0.0 sqlalchemy-1.3.24 sympy-1.11.1 tensorboard-2.11.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 threadpoolctl-3.1.0 tifffile-2022.10.10 tomli-2.0.1 torch-1.11.0+cpu torchio-0.18.86 torchvision-0.12.0+cpu tqdm-4.64.1 transforms3d-0.4.1 typer-0.7.0 types-setuptools-65.6.0.1 typing-extensions-4.4.0 urllib3-1.26.13 urllib3-secure-extra-0.1.0 voxelmorph-0.2 wandb-0.13.5 werkzeug-2.2.2 wheel-0.38.4 win32-setctime-1.1.0 wquantiles-0.6 wrapt-1.14.1 xlwt-1.3.0 zipp-3.11.0
WARNING: You are using pip version 21.1.1; however, version 22.3.1 is available.
You should consider upgrading via the 'c:\users\joshu\repos\spinalcordtoolbox\venv_sct_38\scripts\python.exe -m pip install --upgrade pip' command.
```
</details>

<details>
<summary> Python 3.8.10, pip 22.3.1 (after running --upgrade) </summary>

```console
venv_sct_38_upgraded@C:\Users\joshu\repos\spinalcordtoolbox$ python -m pip install --upgrade pip setuptools wheel
Requirement already satisfied: pip in c:\users\joshu\repos\spinalcordtoolbox\venv_sct_38_upgraded\lib\site-packages (21.1.1)
Collecting pip
  Using cached pip-22.3.1-py3-none-any.whl (2.1 MB)
Requirement already satisfied: setuptools in c:\users\joshu\repos\spinalcordtoolbox\venv_sct_38_upgraded\lib\site-packages (56.0.0)
Collecting setuptools
  Using cached setuptools-65.6.3-py3-none-any.whl (1.2 MB)
Collecting wheel
  Using cached wheel-0.38.4-py3-none-any.whl (36 kB)
Installing collected packages: wheel, setuptools, pip
  Attempting uninstall: setuptools
    Found existing installation: setuptools 56.0.0
    Uninstalling setuptools-56.0.0:
      Successfully uninstalled setuptools-56.0.0
  Attempting uninstall: pip
    Found existing installation: pip 21.1.1
    Uninstalling pip-21.1.1:
      Successfully uninstalled pip-21.1.1
Successfully installed pip-22.3.1 setuptools-65.6.3 wheel-0.38.4

venv_sct_38_upgraded@C:\Users\joshu\repos\spinalcordtoolbox$ pip install -r requirements.txt
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cpu
Collecting dipy
  Using cached dipy-1.5.0-cp38-none-win_amd64.whl (7.4 MB)
Collecting ivadomed~=2.9.6
  Using cached ivadomed-2.9.7-py3-none-any.whl (231 kB)
Collecting matplotlib
  Using cached matplotlib-3.6.2-cp38-cp38-win_amd64.whl (7.2 MB)
Collecting msvc-runtime==14.29.30133
  Using cached msvc_runtime-14.29.30133-cp38-cp38-win_amd64.whl (1.8 MB)
Collecting nibabel
  Using cached nibabel-4.0.2-py3-none-any.whl (3.3 MB)
Collecting nilearn
  Using cached nilearn-0.9.2-py3-none-any.whl (9.6 MB)
Collecting numpy
  Using cached numpy-1.23.5-cp38-cp38-win_amd64.whl (14.7 MB)
Collecting onnxruntime>=1.7.0
  Using cached onnxruntime-1.13.1-cp38-cp38-win_amd64.whl (5.9 MB)
Collecting pandas
  Using cached pandas-1.5.2-cp38-cp38-win_amd64.whl (11.0 MB)
Collecting portalocker
  Using cached portalocker-2.6.0-py2.py3-none-any.whl (15 kB)
Collecting psutil
  Using cached psutil-5.9.4-cp36-abi3-win_amd64.whl (252 kB)
Collecting pyqt5<5.15.0,>=5.12.0
  Using cached PyQt5-5.14.2-5.14.2-cp35.cp36.cp37.cp38-none-win_amd64.whl (52.9 MB)
Collecting pytest
  Using cached pytest-7.2.0-py3-none-any.whl (316 kB)
Collecting pytest-cov
  Using cached pytest_cov-4.0.0-py3-none-any.whl (21 kB)
Collecting raven
  Using cached raven-6.10.0-py2.py3-none-any.whl (284 kB)
Collecting requests
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting requirements-parser
  Using cached requirements_parser-0.5.0-py3-none-any.whl (18 kB)
Collecting scipy
  Using cached scipy-1.9.3-cp38-cp38-win_amd64.whl (39.8 MB)
Collecting scikit-image
  Using cached scikit_image-0.19.3-cp38-cp38-win_amd64.whl (12.2 MB)
Collecting scikit-learn
  Using cached scikit_learn-1.1.3-cp38-cp38-win_amd64.whl (7.5 MB)
Collecting xlwt
  Using cached xlwt-1.3.0-py2.py3-none-any.whl (99 kB)
Collecting tqdm
  Using cached tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
Collecting transforms3d
  Using cached transforms3d-0.4.1.tar.gz (1.4 MB)
  Preparing metadata (setup.py) ... done
Collecting urllib3[secure]
  Using cached urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
Collecting pytest_console_scripts
  Using cached pytest_console_scripts-1.3.1-py3-none-any.whl (9.0 kB)
Collecting pyyaml
  Using cached PyYAML-6.0-cp38-cp38-win_amd64.whl (155 kB)
Collecting voxelmorph
  Using cached voxelmorph-0.2-py3-none-any.whl (54 kB)
Collecting wquantiles
  Using cached wquantiles-0.6-py3-none-any.whl (3.3 kB)
Collecting h5py>=2.8.0
  Downloading h5py-3.7.0-cp38-cp38-win_amd64.whl (2.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.6/2.6 MB 28.4 MB/s eta 0:00:00
Collecting pybids>=0.14.0
  Using cached pybids-0.15.5-py3-none-any.whl (14.9 MB)
Collecting torchvision<=0.12.0,>=0.9.1
  Using cached https://download.pytorch.org/whl/cpu/torchvision-0.12.0%2Bcpu-cp38-cp38-win_amd64.whl (1.0 MB)
Collecting joblib~=1.0
  Using cached joblib-1.2.0-py3-none-any.whl (297 kB)
Collecting tensorboard>=1.15.0
  Using cached tensorboard-2.11.0-py3-none-any.whl (6.0 MB)
Collecting torch<=1.11.0,>=1.8.1
  Using cached https://download.pytorch.org/whl/cpu/torch-1.11.0%2Bcpu-cp38-cp38-win_amd64.whl (157.5 MB)
Collecting wandb>=0.12.11
  Using cached wandb-0.13.5-py2.py3-none-any.whl (1.9 MB)
Collecting pyparsing<3,>=2.0.2
  Using cached pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Collecting loguru~=0.5
  Using cached loguru-0.6.0-py3-none-any.whl (58 kB)
Collecting torchio>=0.18.68
  Using cached torchio-0.18.86-py2.py3-none-any.whl (172 kB)
Collecting pandas
  Using cached pandas-1.4.4-cp38-cp38-win_amd64.whl (10.6 MB)
Collecting nibabel
  Using cached nibabel-3.2.2-py3-none-any.whl (3.3 MB)
Collecting seaborn~=0.11
  Using cached seaborn-0.12.1-py3-none-any.whl (288 kB)
Collecting csv-diff>=1.0
  Using cached csv_diff-1.1-py3-none-any.whl (12 kB)
Collecting imageio~=2.19
  Using cached imageio-2.22.4-py3-none-any.whl (3.4 MB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.4.4-cp38-cp38-win_amd64.whl (55 kB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.38.0-py3-none-any.whl (965 kB)
Collecting contourpy>=1.0.1
  Using cached contourpy-1.0.6-cp38-cp38-win_amd64.whl (163 kB)
Collecting pillow>=6.2.0
  Using cached Pillow-9.3.0-cp38-cp38-win_amd64.whl (2.5 MB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Requirement already satisfied: setuptools in c:\users\joshu\repos\spinalcordtoolbox\venv_sct_38_upgraded\lib\site-packages (from nibabel->-r requirements.txt (line 18)) (65.6.3)
Collecting lxml
  Using cached lxml-4.9.1-cp38-cp38-win_amd64.whl (3.6 MB)
Collecting coloredlogs
  Using cached coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
Collecting flatbuffers
  Using cached flatbuffers-22.11.23-py2.py3-none-any.whl (26 kB)
Collecting sympy
  Using cached sympy-1.11.1-py3-none-any.whl (6.5 MB)
Collecting protobuf
  Using cached protobuf-4.21.9-cp38-cp38-win_amd64.whl (525 kB)
Collecting pytz>=2020.1
  Using cached pytz-2022.6-py2.py3-none-any.whl (498 kB)
Collecting pywin32>=226
  Using cached pywin32-305-cp38-cp38-win_amd64.whl (12.3 MB)
Collecting PyQt5-sip<13,>=12.7
  Using cached PyQt5_sip-12.11.0-cp38-cp38-win_amd64.whl (78 kB)
Collecting tomli>=1.0.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting colorama
  Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting exceptiongroup>=1.0.0rc8
  Using cached exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting attrs>=19.2.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting coverage[toml]>=5.2.1
  Using cached coverage-6.5.0-cp38-cp38-win_amd64.whl (188 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting types-setuptools>=57.0.0
  Using cached types_setuptools-65.6.0.1-py3-none-any.whl (48 kB)
Collecting tifffile>=2019.7.26
  Using cached tifffile-2022.10.10-py3-none-any.whl (210 kB)
Collecting PyWavelets>=1.1.1
  Using cached PyWavelets-1.4.1-cp38-cp38-win_amd64.whl (4.2 MB)
Collecting networkx>=2.2
  Using cached networkx-2.8.8-py3-none-any.whl (2.0 MB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting cryptography>=1.3.4
  Using cached cryptography-38.0.4-cp36-abi3-win_amd64.whl (2.4 MB)
Collecting pyOpenSSL>=0.14
  Using cached pyOpenSSL-22.1.0-py3-none-any.whl (57 kB)
Collecting urllib3-secure-extra
  Using cached urllib3_secure_extra-0.1.0-py2.py3-none-any.whl (1.4 kB)
Collecting neurite>=0.2
  Using cached neurite-0.2-py3-none-any.whl (108 kB)
Collecting cffi>=1.12
  Using cached cffi-1.15.1-cp38-cp38-win_amd64.whl (178 kB)
Collecting click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting dictdiffer
  Using cached dictdiffer-0.9.0-py2.py3-none-any.whl (16 kB)
Collecting win32-setctime>=1.0.0
  Using cached win32_setctime-1.1.0-py3-none-any.whl (3.6 kB)
Collecting pystrum>=0.2
  Using cached pystrum-0.2-py3-none-any.whl (19 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting formulaic<0.4,>=0.2.4
  Using cached formulaic-0.3.4-py3-none-any.whl (68 kB)
Collecting sqlalchemy<1.4.0.dev0
  Using cached SQLAlchemy-1.3.24-cp38-cp38-win_amd64.whl (1.2 MB)
Collecting num2words
  Using cached num2words-0.5.12-py3-none-any.whl (125 kB)
Collecting bids-validator
  Using cached bids_validator-1.9.9-py2.py3-none-any.whl (21 kB)
Collecting tensorboard-plugin-wit>=1.6.0
  Using cached tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)
Collecting google-auth<3,>=1.6.3
  Using cached google_auth-2.14.1-py2.py3-none-any.whl (175 kB)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting grpcio>=1.24.3
  Using cached grpcio-1.50.0-cp38-cp38-win_amd64.whl (3.7 MB)
Requirement already satisfied: wheel>=0.26 in c:\users\joshu\repos\spinalcordtoolbox\venv_sct_38_upgraded\lib\site-packages (from tensorboard>=1.15.0->ivadomed~=2.9.6->-r requirements.txt (line 9)) (0.38.4)
Collecting absl-py>=0.4
  Using cached absl_py-1.3.0-py3-none-any.whl (124 kB)
Collecting protobuf
  Using cached protobuf-3.20.3-cp38-cp38-win_amd64.whl (904 kB)
Collecting werkzeug>=1.0.1
  Using cached Werkzeug-2.2.2-py3-none-any.whl (232 kB)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-any.whl (2.4 kB)
Collecting markdown>=2.6.8
  Using cached Markdown-3.4.1-py3-none-any.whl (93 kB)
Collecting typing-extensions
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting Deprecated
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting SimpleITK!=2.0.*,!=2.1.1.1
  Using cached SimpleITK-2.2.0-cp38-cp38-win_amd64.whl (27.1 MB)
Collecting humanize
  Using cached humanize-4.4.0-py3-none-any.whl (106 kB)
Collecting typer[all]
  Using cached typer-0.7.0-py3-none-any.whl (38 kB)
Collecting docker-pycreds>=0.4.0
  Using cached docker_pycreds-0.4.0-py2.py3-none-any.whl (9.0 kB)
Collecting shortuuid>=0.5.0
  Using cached shortuuid-1.0.11-py3-none-any.whl (10 kB)
Collecting setproctitle
  Using cached setproctitle-1.3.2-cp38-cp38-win_amd64.whl (11 kB)
Collecting pathtools
  Using cached pathtools-0.1.2.tar.gz (11 kB)
  Preparing metadata (setup.py) ... done
Collecting GitPython>=1.0.0
  Using cached GitPython-3.1.29-py3-none-any.whl (182 kB)
Collecting promise<3,>=2.0
  Using cached promise-2.3.tar.gz (19 kB)
  Preparing metadata (setup.py) ... done
Collecting sentry-sdk>=1.0.0
  Using cached sentry_sdk-1.11.1-py2.py3-none-any.whl (168 kB)
Collecting humanfriendly>=9.1
  Using cached humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
Collecting mpmath>=0.19
  Using cached mpmath-1.2.1-py3-none-any.whl (532 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting astor>=0.8
  Using cached astor-0.8.1-py2.py3-none-any.whl (27 kB)
Collecting interface-meta<2.0.0,>=1.2.0
  Using cached interface_meta-1.3.0-py3-none-any.whl (14 kB)
Collecting wrapt>=1.0
  Using cached wrapt-1.14.1-cp38-cp38-win_amd64.whl (35 kB)
Collecting gitdb<5,>=4.0.1
  Using cached gitdb-4.0.10-py3-none-any.whl (62 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting requests-oauthlib>=0.7.0
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting pyreadline3
  Using cached pyreadline3-3.4.1-py3-none-any.whl (95 kB)
Collecting importlib-metadata>=4.4
  Using cached importlib_metadata-5.1.0-py3-none-any.whl (21 kB)
Collecting MarkupSafe>=2.1.1
  Using cached MarkupSafe-2.1.1-cp38-cp38-win_amd64.whl (17 kB)
Collecting docopt>=0.6.2
  Using cached docopt-0.6.2.tar.gz (25 kB)
  Preparing metadata (setup.py) ... done
Collecting shellingham<2.0.0,>=1.3.0
  Using cached shellingham-1.5.0-py2.py3-none-any.whl (9.3 kB)
Collecting rich<13.0.0,>=10.11.0
  Using cached rich-12.6.0-py3-none-any.whl (237 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting zipp>=0.5
  Using cached zipp-3.11.0-py3-none-any.whl (6.6 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Collecting commonmark<0.10.0,>=0.9.0
  Using cached commonmark-0.9.1-py2.py3-none-any.whl (51 kB)
Collecting pygments<3.0.0,>=2.6.0
  Using cached Pygments-2.13.0-py3-none-any.whl (1.1 MB)
Building wheels for collected packages: transforms3d, promise, pathtools, docopt
  Building wheel for transforms3d (setup.py) ... done
  Created wheel for transforms3d: filename=transforms3d-0.4.1-py3-none-any.whl size=1376797 sha256=147a791000251889594a623e78734ac68e0645dfaf2c0acb653db67c64b8cd94
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\ef\da\58\f08940e8ea16b7b4f2c63b030475e935bc4de01547f32f3a0f
  Building wheel for promise (setup.py) ... done
  Created wheel for promise: filename=promise-2.3-py3-none-any.whl size=21548 sha256=ed18d96e4472caa5b494f89a2d7b4dbcddda6c7cc12ab05cbc5a03f62cdb4d61
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\6a\fe\dc\a7b3e03dfd0afb3a19691905bbafac1fbaebb704a02a4daeb2
  Building wheel for pathtools (setup.py) ... done
  Created wheel for pathtools: filename=pathtools-0.1.2-py3-none-any.whl size=8801 sha256=b7a5c1e130bc1b08938472121ba0b11ff43bff1f139b0fe4606e4c9d2ea7ff64
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\4d\33\74\7c0903053e955973d5dc3d21857a29b3f8c0806ad0b05c32a1
  Building wheel for docopt (setup.py) ... done
  Created wheel for docopt: filename=docopt-0.6.2-py2.py3-none-any.whl size=13775 sha256=67ab39b231b2afc5f21e524fece94b63ff534b70f9596d087ee79632aef6aefa
  Stored in directory: c:\users\joshu\appdata\local\pip\cache\wheels\ca\cc\e3\f1e272f628fdb013d969acc99cfe2e031ea15b3efb74ffe842
Successfully built transforms3d promise pathtools docopt
Installing collected packages: xlwt, urllib3-secure-extra, types-setuptools, tensorboard-plugin-wit, SimpleITK, raven, pywin32, pytz, pyreadline3, pyasn1, pathtools, mpmath, iniconfig, flatbuffers, docopt, dictdiffer, commonmark, bids-validator, zipp, wrapt, win32-setctime, urllib3, typing-extensions, transforms3d, tomli, threadpoolctl, tensorboard-data-server, sympy, sqlalchemy, smmap, six, shortuuid, shellingham, setproctitle, rsa, requirements-parser, pyyaml, PyQt5-sip, pyparsing, pygments, pycparser, pyasn1-modules, psutil, protobuf, portalocker, pluggy, pillow, oauthlib, numpy, num2words, networkx, msvc-runtime, MarkupSafe, lxml, kiwisolver, joblib, interface-meta, idna, humanize, humanfriendly, fonttools, exceptiongroup, cycler, coverage, colorama, charset-normalizer, certifi, cachetools, attrs, astor, absl-py, wquantiles, werkzeug, tqdm, torch, tifffile, sentry-sdk, scipy, rich, requests, PyWavelets, python-dateutil, pyqt5, promise, packaging, loguru, importlib-metadata, imageio, h5py, grpcio, google-auth, gitdb, docker-pycreds, Deprecated, contourpy, coloredlogs, click, cffi, typer, torchvision, scikit-learn, scikit-image, requests-oauthlib, pytest, pandas, onnxruntime, nibabel, matplotlib, markdown, GitPython, csv-diff, cryptography, wandb, seaborn, pytest-cov, pytest_console_scripts, pystrum, pyOpenSSL, nilearn, google-auth-oauthlib, formulaic, dipy, torchio, tensorboard, pybids, neurite, voxelmorph, ivadomed
Successfully installed Deprecated-1.2.13 GitPython-3.1.29 MarkupSafe-2.1.1 PyQt5-sip-12.11.0 PyWavelets-1.4.1 SimpleITK-2.2.0 absl-py-1.3.0 astor-0.8.1 attrs-22.1.0 bids-validator-1.9.9 cachetools-5.2.0 certifi-2022.9.24 cffi-1.15.1 charset-normalizer-2.1.1 click-8.1.3 colorama-0.4.6 coloredlogs-15.0.1 commonmark-0.9.1 contourpy-1.0.6 coverage-6.5.0 cryptography-38.0.4 csv-diff-1.1 cycler-0.11.0 dictdiffer-0.9.0 dipy-1.5.0 docker-pycreds-0.4.0 docopt-0.6.2 exceptiongroup-1.0.4 flatbuffers-22.11.23 fonttools-4.38.0 formulaic-0.3.4 gitdb-4.0.10 google-auth-2.14.1 google-auth-oauthlib-0.4.6 grpcio-1.50.0 h5py-3.7.0 humanfriendly-10.0 humanize-4.4.0 idna-3.4 imageio-2.22.4 importlib-metadata-5.1.0 iniconfig-1.1.1 interface-meta-1.3.0 ivadomed-2.9.7 joblib-1.2.0 kiwisolver-1.4.4 loguru-0.6.0 lxml-4.9.1 markdown-3.4.1 matplotlib-3.6.2 mpmath-1.2.1 msvc-runtime-14.29.30133 networkx-2.8.8 neurite-0.2 nibabel-3.2.2 nilearn-0.9.2 num2words-0.5.12 numpy-1.23.5 oauthlib-3.2.2 onnxruntime-1.13.1 packaging-21.3 pandas-1.4.4 pathtools-0.1.2 pillow-9.3.0 pluggy-1.0.0 portalocker-2.6.0 promise-2.3 protobuf-3.20.3 psutil-5.9.4 pyOpenSSL-22.1.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pybids-0.15.5 pycparser-2.21 pygments-2.13.0 pyparsing-2.4.7 pyqt5-5.14.2 pyreadline3-3.4.1 pystrum-0.2 pytest-7.2.0 pytest-cov-4.0.0 pytest_console_scripts-1.3.1 python-dateutil-2.8.2 pytz-2022.6 pywin32-305 pyyaml-6.0 raven-6.10.0 requests-2.28.1 requests-oauthlib-1.3.1 requirements-parser-0.5.0 rich-12.6.0 rsa-4.9 scikit-image-0.19.3 scikit-learn-1.1.3 scipy-1.9.3 seaborn-0.12.1 sentry-sdk-1.11.1 setproctitle-1.3.2 shellingham-1.5.0 shortuuid-1.0.11 six-1.16.0 smmap-5.0.0 sqlalchemy-1.3.24 sympy-1.11.1 tensorboard-2.11.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 threadpoolctl-3.1.0 tifffile-2022.10.10 tomli-2.0.1 torch-1.11.0+cpu torchio-0.18.86 torchvision-0.12.0+cpu tqdm-4.64.1 transforms3d-0.4.1 typer-0.7.0 types-setuptools-65.6.0.1 typing-extensions-4.4.0 urllib3-1.26.13 urllib3-secure-extra-0.1.0 voxelmorph-0.2 wandb-0.13.5 werkzeug-2.2.2 win32-setctime-1.1.0 wquantiles-0.6 wrapt-1.14.1 xlwt-1.3.0 zipp-3.11.0
```
</details>

It looks like the default, non-upgraded version of pip for Python 3.7.9 (20.1.1) displays the following message:

```
ERROR: tensorboard 2.11.0 has requirement protobuf<4,>=3.9.2, but you'll have protobuf 4.21.9 which is incompatible.
ERROR: ivadomed 2.9.7 has requirement nibabel~=3.2, but you'll have nibabel 4.0.2 which is incompatible.
```

However, the installation doesn't actually... _fail_?

And, it looks like upgrading to Python 3.8 (pip 21.1.1) causes these errors to disappear anyway. So, it seems like the upgrade is non-critical, and could probably be removed.

That said, it may be worthwhile to keep the upgrade step, if only to remove this ominous warning at the end of the installation:

```
WARNING: You are using pip version 21.1.1; however, version 22.3.1 is available.
You should consider upgrading via the 'c:\users\joshu\repos\spinalcordtoolbox\venv_sct_38\scripts\python.exe -m pip install --upgrade pip' command.
```

We can just change the `p3.7`-specific comment into one that describes the warning message above.

Reply:
> That said, it may be worthwhile to keep the upgrade step, if only to remove this ominous warning at the end of the installation:

I think making sure `pip` is up-to-date before using it to install other packages is a common pattern, so I'd say let's keep it.

> Still, if we do decide to keep the upgrade step, we may also want to add a "pip!=21.2.*" clause, just to keep things consistent across installations for different OSs.

I agree! I prefer being consistent.